Panic when query contains `order` and `_version`

E.g. The below works:
```
Log (filter: { _and: [ { blockNumber: { _ge: 1 } }, { blockNumber: { _le: 1000 } } ] }){
  _docID
 _version { cid signature { type identity value } } 
 address
 topics 
 data
 transactionHash
 blockNumber
}
```
But the following panics:
```
Log (filter: { _and: [ { blockNumber: { _ge: 1 } }, { blockNumber: { _le: 1000 } } ] }, order: { blockNumber: DESC }){
  _docID
  _version { cid signature { type identity value } } 
  address
  topics
  data
  transactionHash
  blockNumber
}
```
With:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x10930b9ac]

goroutine 1269 [running]:
github.com/sourcenetwork/defradb/internal/planner.(*multiScanNode).Source(0x0)
	/Users/quinnpurdy/go/pkg/mod/github.com/sourcenetwork/defradb@v0.20.0/internal/planner/scan.go:407 +0x1c
github.com/sourcenetwork/defradb/internal/planner.getNode[...]({0x10cbbba60, 0x1400160e000})
	/Users/quinnpurdy/go/pkg/mod/github.com/sourcenetwork/defradb@v0.20.0/internal/planner/type_join.go:853 +0xd0
github.com/sourcenetwork/defradb/internal/planner.isOrderedByIndex({0x10cbbba60, 0x1400160e000})
	/Users/quinnpurdy/go/pkg/mod/github.com/sourcenetwork/defradb@v0.20.0/internal/planner/planner.go:322 +0x44
github.com/sourcenetwork/defradb/internal/planner.(*Planner).expandSelectTopNodePlan(0x14012111c70, 0x1401215d810, 0x0)
	/Users/quinnpurdy/go/pkg/mod/github.com/sourcenetwork/defradb@v0.20.0/internal/planner/planner.go:240 +0x1bc
github.com/sourcenetwork/defradb/internal/planner.(*Planner).expandPlan(0x14012111c70, {0x10cbbb0a0, 0x1401215d810}, 0x0)
	/Users/quinnpurdy/go/pkg/mod/github.com/sourcenetwork/defradb@v0.20.0/internal/planner/planner.go:154 +0x4b8
github.com/sourcenetwork/defradb/internal/planner.(*Planner).expandMultiNode(0x14012111c70, {0x119ebbe20, 0x14012147040}, 0x0)
	/Users/quinnpurdy/go/pkg/mod/github.com/sourcenetwork/defradb@v0.20.0/internal/planner/planner.go:276 +0xa8
github.com/sourcenetwork/defradb/internal/planner.(*Planner).expandPlan(0x14012111c70, {0x10cbbb700, 0x14012147040}, 0x0)
	/Users/quinnpurdy/go/pkg/mod/github.com/sourcenetwork/defradb@v0.20.0/internal/planner/planner.go:191 +0x6f8
github.com/sourcenetwork/defradb/internal/planner.(*Planner).optimizePlan(0x14012111c70, {0x10cbbb700, 0x14012147040})
	/Users/quinnpurdy/go/pkg/mod/github.com/sourcenetwork/defradb@v0.20.0/internal/planner/planner.go:146 +0x40
github.com/sourcenetwork/defradb/internal/planner.(*Planner).MakePlan(0x14012111c70, 0x14012146fa0)
	/Users/quinnpurdy/go/pkg/mod/github.com/sourcenetwork/defradb@v0.20.0/internal/planner/planner.go:726 +0x21c
github.com/sourcenetwork/defradb/internal/planner.(*Planner).RunRequest(0x14012111c70, {0x10cb90370, 0x1400ca66c60}, 0x14012146fa0)
	/Users/quinnpurdy/go/pkg/mod/github.com/sourcenetwork/defradb@v0.20.0/internal/planner/planner.go:639 +0x58
github.com/sourcenetwork/defradb/internal/db.(*DB).execRequest(0x14000271600, {0x10cb90370, 0x1400ca66c60}, {0x1400031be00, 0xea}, 0x1400ca31170)
	/Users/quinnpurdy/go/pkg/mod/github.com/sourcenetwork/defradb@v0.20.0/internal/db/request.go:59 +0x654
github.com/sourcenetwork/defradb/internal/db.(*DB).ExecRequest(0x14000271600, {0x10cb90370, 0x1400ca66c60}, {0x1400031be00, 0xea}, {0x0, 0x0, 0x0})
	/Users/quinnpurdy/go/pkg/mod/github.com/sourcenetwork/defradb@v0.20.0/internal/db/store.go:42 +0x434
github.com/shinzonetwork/app-sdk/pkg/defra.(*queryClient).query(0x14001971b88, {0x10cb903a8, 0x140016645a0}, {0x1400031be00, 0xea})
	/Users/quinnpurdy/go/pkg/mod/github.com/shinzonetwork/app-sdk@v0.0.0-20251107185640-33925fbdeafc/pkg/defra/query.go:35 +0xdc
github.com/shinzonetwork/app-sdk/pkg/defra.(*queryClient).queryDataInto(0x14001971b88, {0x10cb903a8, 0x140016645a0}, {0x1400031be00, 0xea}, {0x10bed33c0, 0x1400ca310e0})
	/Users/quinnpurdy/go/pkg/mod/github.com/shinzonetwork/app-sdk@v0.0.0-20251107185640-33925fbdeafc/pkg/defra/query.go:96 +0x6c
github.com/shinzonetwork/app-sdk/pkg/defra.QueryArray[...]({0x10cb903a8, 0x140016645a0}, 0x14000657b90, {0x1400ca76000, 0xe0})
	/Users/quinnpurdy/go/pkg/mod/github.com/shinzonetwork/app-sdk@v0.0.0-20251107185640-33925fbdeafc/pkg/defra/query.go:234 +0x170
github.com/shinzonetwork/shinzo-host-client/pkg/host.(*Host).ApplyView(0x1400781db00, {0x10cb903a8, 0x140016645a0}, {{0x1400214a085, 0x61}, 0x1400208a1e0, 0x1400208a1f0, {{0x14002089a10, 0x1, 0x1}}, ...}, ...)
	/Users/quinnpurdy/Documents/repos/host2/pkg/host/viewHosting.go:111 +0x30c
github.com/shinzonetwork/shinzo-host-client/pkg/host.(*Host).processView(0x1400781db00, {0x10cb903a8, 0x140016645a0}, 0x14007849d60)
	/Users/quinnpurdy/Documents/repos/host2/pkg/host/blockProcessing.go:75 +0x328
github.com/shinzonetwork/shinzo-host-client/pkg/host.(*Host).processAllViews(0x1400781db00, {0x10cb903a8, 0x140016645a0})
	/Users/quinnpurdy/Documents/repos/host2/pkg/host/blockProcessing.go:98 +0x1fc
created by github.com/shinzonetwork/shinzo-host-client/pkg/host.StartHostingWithEventSubscription in goroutine 6
	/Users/quinnpurdy/Documents/repos/host2/pkg/host/host.go:98 +0xa40
```

--- a/internal/planner/multi.go
+++ b/internal/planner/multi.go
@@ -63,6 +63,7 @@ type parallelNode struct { // serialNode?
 	children     []planNode
 	childIndexes []int
 
+	source    planNode
 	multiscan *multiScanNode
 }
 
@@ -181,7 +182,7 @@ func (p *parallelNode) nextAppend(index int, plan planNode) (bool, error) {
 	return true, nil
 }
 
-func (p *parallelNode) Source() planNode { return p.multiscan }
+func (p *parallelNode) Source() planNode { return p.source }
 
 func (p *parallelNode) Children() []planNode {
 	return p.children
@@ -202,6 +203,7 @@ func (n *selectNode) addSubPlan(fieldIndex int, newPlan planNode) error {
 		case *dagScanNode:
 			m := &parallelNode{
 				p:         n.planner,
+				source:    newPlan,
 				docMapper: docMapper{n.source.DocumentMap()},
 			}
 			m.addChild(-1, n.source)
@@ -226,6 +228,7 @@ func (n *selectNode) addSubPlan(fieldIndex int, newPlan planNode) error {
 		parallelNode := &parallelNode{
 			p:         n.planner,
 			multiscan: multiscan,
+			source:    multiscan,
 			docMapper: docMapper{n.source.DocumentMap()},
 		}
 		parallelNode.addChild(-1, n.source)

--- a/tests/integration/query/simple/with_version_order_test.go
+++ b/tests/integration/query/simple/with_version_order_test.go
@@ -1,0 +1,74 @@
+// Copyright 2025 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package simple
+
+import (
+	"testing"
+
+	"github.com/sourcenetwork/defradb/tests/action"
+	testUtils "github.com/sourcenetwork/defradb/tests/integration"
+)
+
+func TestQuerySimpleWithVersionAndOrder(t *testing.T) {
+	test := testUtils.TestCase{
+		Actions: []any{
+			&action.AddSchema{
+				Schema: `
+					type Users {
+						Name: String
+					}
+				`,
+			},
+			testUtils.CreateDoc{
+				DocMap: map[string]any{
+					"Name": "John",
+				},
+			},
+			testUtils.CreateDoc{
+				DocMap: map[string]any{
+					"Name": "Chris",
+				},
+			},
+			testUtils.Request{
+				Request: `query {
+					Users(order: {Name: ASC}) {
+						Name
+						_version {
+							fieldName
+						}
+					}
+				}`,
+				Results: map[string]any{
+					"Users": []map[string]any{
+						{
+							"Name": "Chris",
+							"_version": []map[string]any{
+								{
+									"fieldName": "_C",
+								},
+							},
+						},
+						{
+							"Name": "John",
+							"_version": []map[string]any{
+								{
+									"fieldName": "_C",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	testUtils.ExecuteTestCase(t, test)
+}

--- a/tests/integration/query/simple/with_version_order_test.go
+++ b/tests/integration/query/simple/with_version_order_test.go
@@ -23,24 +23,24 @@ func TestQuerySimpleWithVersionAndOrder(t *testing.T) {
 			&action.AddSchema{
 				Schema: `
 					type Users {
-						Name: String
+						name: String
 					}
 				`,
 			},
 			testUtils.CreateDoc{
 				DocMap: map[string]any{
-					"Name": "John",
+					"name": "John",
 				},
 			},
 			testUtils.CreateDoc{
 				DocMap: map[string]any{
-					"Name": "Chris",
+					"name": "Chris",
 				},
 			},
 			testUtils.Request{
 				Request: `query {
-					Users(order: {Name: ASC}) {
-						Name
+					Users(order: {name: ASC}) {
+						name
 						_version {
 							fieldName
 						}
@@ -49,7 +49,7 @@ func TestQuerySimpleWithVersionAndOrder(t *testing.T) {
 				Results: map[string]any{
 					"Users": []map[string]any{
 						{
-							"Name": "Chris",
+							"name": "Chris",
 							"_version": []map[string]any{
 								{
 									"fieldName": "_C",
@@ -57,7 +57,7 @@ func TestQuerySimpleWithVersionAndOrder(t *testing.T) {
 							},
 						},
 						{
-							"Name": "John",
+							"name": "John",
 							"_version": []map[string]any{
 								{
 									"fieldName": "_C",

